zebra-consensus/src/checkpoint.rs

```diff
@@ -327,6 +328,9 @@ impl CheckpointVerifier {
         // Add the block to the list of queued blocks at this height
         let hash = block.as_ref().into();
         let new_qblock = QueuedBlock { block, hash, tx };
+        // We avoid allocating extra space, because multiple blocks are rare,
+        // and block headers are large.
```

Comment:
I don't think this changes whether it's a good idea to use `with_capacity(1)`, but since `block` is an `Arc<Block>`, it's just the size of a single pointer, not the size of a full block header.

Reply:
You're right, I was still thinking of Blocks.

I'll just revert that part of the change then, there's no need for us to worry about 8 bytes vs 32 bytes. We might as well just allocate the 32 bytes upfront. (And that will reduce memory fragmentation.)

Reply:
Sorry I was unclear, I think it's still a good idea to use `with_capacity(1)` -- it's the common case and it means we'll get exactly the right size in the common case.

Reply:
Done in 095a37f9462eb185fb74c6a76457949d08606bde

Reply:
Ah right now I understand. I fixed it properly in 002e148616e5d75db77d01b962371483b3129537.

Reply:
>  002

*No I am not mining commit hashes, why do you ask?* 😃